api/explorer/api/ingest.py

```diff
@@ -210,30 +210,61 @@ def write_edge(input: Tuple[graph.Package, graph.Edge, graph.Package]):
         )
 
     with ThreadPoolExecutor() as ex:
-        click.echo("Writing packages")
+        # Write nodes to the Gremlin graph
+        click.echo("Writing nodes to the Gremlin graph...")
         with click.progressbar(length=len(packages)) as bar:
             futures = [ex.submit(write_package, p) for p in packages]
             for f in futures:
                 f.add_done_callback(lambda _x: bar.update(1))
             wait(futures)
-            exceptions = [f.exception() for f in futures if f.exception()]
-            if exceptions:
-                # Raise a single exception after all tasks have completed
-                raise Exception(
-                    "Exceptions occurred while writing packages."
-                ) from exceptions[0]
-
-        click.echo("Writing edges")
+            # Collect the packages which are not ingested successfully
+            failed_packages = [
+                packages[futures.index(f)] for f in futures if f.exception()
+            ]
+            failed_packages_copy = failed_packages.copy()
+            if failed_packages_copy:
+                click.echo("\nRetrying to write nodes...")
+                for i in range(len(failed_packages_copy)):
+                    try:
+                        graph.insert_unique_vertex(failed_packages_copy[i], g_factory())
+                        failed_packages.remove(failed_packages_copy[i])
+                    except Exception as e:
+                        logger.exception(f"An exception occurred: {e}")
```

Comment:
```suggestion
                        logger.error(f"Synchronous insertion failed for node: {failed_package_copy[i]}")
                        logger.exception(e)
```

https://docs.python.org/3/library/logging.html
> If exc_info does not evaluate as false, it causes exception information to be added to the logging message. If an exception tuple (in the format returned by [sys.exc_info()](https://docs.python.org/3/library/sys.html#sys.exc_info)) or an exception instance is provided, it is used; otherwise, [sys.exc_info()](https://docs.python.org/3/library/sys.html#sys.exc_info) is called to get the exception information.